components/channel_layout/channel_controller.jsx

```diff
@@ -22,16 +22,51 @@ import ModalController from 'components/modal_controller';
 import TeamSidebar from 'components/team_sidebar';
 import Sidebar from 'components/sidebar';
 import * as Utils from 'utils/utils';
+import * as UserAgent from 'utils/user_agent.jsx';
 import CenterChannel from 'components/channel_layout/center_channel';
+import LoadingScreen from 'components/loading_screen';
 
 export default class ChannelController extends React.Component {
     static propTypes = {
         pathName: PropTypes.string.isRequired,
         teamType: PropTypes.string.isRequired,
+        fetchingChannels: PropTypes.bool.isRequired,
     };
 
     shouldComponentUpdate(nextProps) {
-        return this.props.teamType !== nextProps.teamType || this.props.pathName !== nextProps.pathName;
+        return this.props.teamType !== nextProps.teamType || this.props.pathName !== nextProps.pathName || this.props.fetchingChannels !== nextProps.fetchingChannels;
+    }
+
+    componentDidMount() {
+        const platform = window.navigator.platform;
+        const elements = document.getElementsByTagName('body');
+        if (elements.length !== 1) {
+            return;
+        }
+        const body = elements[0];
+
+        body.classList.add('app__body');
+
+        // IE Detection
+        if (UserAgent.isInternetExplorer() || UserAgent.isEdge()) {
+            body.classList.add('browser--ie');
+        }
```

Comment:
Is it worth adding `browser--ie` (and `os--windows` and `os--mac`) from a more global location and have the styling target it with `.app_body.browser--ie` (or something) so that it doesn't affect existing pages? This just makes the 3rd place that `browser--ie` is being added and it would be a shame if that number keeps growing. Just wanted to brainstorm it really.

Reply:
This is just moving code from one place to another, and is part of cherry-pick PR for 5.16, I'm ok to discuss this as part of another Jira issue, but I think for now, if the code looks good (in the sense of fix the problem and keep the old existing correct behavior), we can merge it and talk about the `browser--ie` class as a separate thing.

Reply:
Totally fair.